baremaps-core/src/main/java/org/apache/baremaps/workflow/tasks/CreateIplocIndex.java

```diff
@@ -26,101 +26,93 @@
+
+        logger.info("Generating NIC objects stream");
+        Arrays.stream(nicPaths).parallel().forEach(path -> {
+          try (InputStream inputStream = new BufferedInputStream(Files.newInputStream(Path.of(path)));) {
+            var nicObjects = NicParser.parse(inputStream);
+            logger.info("Inserting the nic objects into the Iploc database");
+            ipLoc.insertNicObjects(nicObjects);
           } catch (IOException e) {
-            throw new UncheckedIOException(e);
+            throw new StreamException(e);
           }
         });
-      } catch (IOException e) {
-        throw new UncheckedIOException(e);
-      }
-    });
-
-      logger.info("Creating the Iploc database");
-      String jdbcUrl = String.format("JDBC:sqlite:%s", targetIplocIndexPath);
-
-      SqliteUtils.executeResource(jdbcUrl, "iploc_init.sql");
-
-      logger.info("Inserting the nic objects into the Iploc database");
-      org.apache.baremaps.iploc.IpLoc ipLoc = new IpLoc(jdbcUrl, geocoder);
-      ipLoc.insertNicObjects(fetchNicObjectStream);
-      IpLocStats ipLocStats = ipLoc.getIplocStats();
-
-      logger.info(
-              """
-                      IpLoc stats
-                      -----------
-                      inetnumInsertedByAddress : {}
-                      inetnumInsertedByDescr : {}
-                      inetnumInsertedByCountry : {}
-                      inetnumInsertedByCountryCode : {}
-                      inetnumInsertedByGeoloc : {}
-                      inetnumNotInserted : {}""",
-              ipLocStats.getInsertedByAddressCount(), ipLocStats.getInsertedByDescrCount(),
-              ipLocStats.getInsertedByCountryCount(), ipLocStats.getInsertedByCountryCodeCount(),
-              ipLocStats.getInsertedByGeolocCount(), ipLocStats.getNotInsertedCount());
-
-      logger.info("IpLoc database created successfully");
-
-    logger.info("Finished creating the Geocoder index {}", targetIplocIndexPath);
-  }
-
-  @Override
-  public boolean equals(Object o) {
-    if (this == o) return true;
-    if (o == null || getClass() != o.getClass()) return false;
-    CreateIplocIndex that = (CreateIplocIndex) o;
-    return Objects.equals(geonamesIndexPath, that.geonamesIndexPath) && Arrays.equals(iplocNicPath, that.iplocNicPath) && Objects.equals(targetIplocIndexPath, that.targetIplocIndexPath);
-  }
-
-  @Override
-  public int hashCode() {
-    int result = Objects.hash(geonamesIndexPath, targetIplocIndexPath);
-    result = 31 * result + Arrays.hashCode(iplocNicPath);
-    return result;
-  }
-
-  @Override
-  public String toString() {
-    return "CreateIplocIndex{" +
-            "geonamesIndexPath='" + geonamesIndexPath + '\'' +
-            ", iplocNicPath=" + Arrays.toString(iplocNicPath) +
-            ", targetIplocIndexPath='" + targetIplocIndexPath + '\'' +
-            '}';
-  }
+
+        IpLocStats ipLocStats = ipLoc.getIplocStats();
+
+        logger.info(
```

Comment:
I think it is really good to do some reporting, we may want to generalize this for all the workflow tasks. Including the time taken, number of items inserted, etc. What do you think?

Reply:
I agree. I am not 100% sure how to make something that is common for every Workflow though. I created ticket #556.